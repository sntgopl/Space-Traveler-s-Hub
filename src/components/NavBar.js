import { NavLink } from 'react-router-dom';
import image from '../images/planet.png';
import '../stylesheets/NavBar.css';

const NavBar = () => (
  <nav>
    <div className="navTitle">
      <img src={image} alt="planet" />
      <h1>Space Traveler`s Hub</h1>
    </div>
    <div className="navList">
      <ul>
        <NavLink className={({ isActive }) => (isActive ? 'activeLink' : '')} to="/">Rockets</NavLink>
        <NavLink className={({ isActive }) => (isActive ? 'activeLink' : '')} to="/missions">Missions</NavLink>
        <NavLink className={({ isActive }) => (isActive ? 'activeLink' : '')} to="/profile">My Profile</NavLink>
      </ul>
    </div>
  </nav>
);

export default NavBar;
