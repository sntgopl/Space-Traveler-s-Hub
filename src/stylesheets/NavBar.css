nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 90%;
  margin: auto;
  border-bottom: 1px solid #ccc;
}

.navTitle {
  display: flex;
  align-items: center;
  gap: 10px;
  margin: 10px 0;
}

.navTitle img {
  width: 70px;
}

.navTitle h1 {
  margin: 0;
}

.navList ul {
  display: flex;
  margin: 0;
}

.navList ul :nth-child(even) {
  border-right: 1px solid #000;
}

.navList ul > * {
  text-decoration: none;
  padding: 0 10px;
}

.activeLink {
  opacity: 0.5;
  border-bottom: 1px solid;
}
