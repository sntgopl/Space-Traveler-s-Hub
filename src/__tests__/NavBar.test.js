import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavBar from '../components/NavBar';

describe('test snapshot', () => {
  it('should render Rocket Container', () => {
    const renderRocketContainer = renderer.create(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    ).toJSON();
    expect(renderRocketContainer).toMatchSnapshot();
  });
});
