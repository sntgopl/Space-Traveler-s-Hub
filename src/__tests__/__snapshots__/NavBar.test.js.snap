// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot should render Rocket Container 1`] = `
<nav>
  <div
    className="navTitle"
  >
    <img
      alt="planet"
      src="planet.png"
    />
    <h1>
      Space Traveler\`s Hub
    </h1>
  </div>
  <div
    className="navList"
  >
    <ul>
      <a
        aria-current="page"
        className="activeLink"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
      <a
        className=""
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
      <a
        className=""
        href="/profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </ul>
  </div>
</nav>
`;
